azkaban-exec-server/src/main/java/azkaban/container/FlowContainer.java

```diff
@@ -0,0 +1,305 @@
+import java.util.concurrent.Future;
+import java.util.zip.ZipFile;
+import javax.sql.DataSource;
+import org.apache.commons.dbutils.QueryRunner;
+import org.apache.log4j.Logger;
+
+/**
+ *  This class is responsible for launching a flow in a container.
+ */
+public class FlowContainer {
+
+  private static final String PROJECT_DIR = "project";
+  private static final String JOBTYPE_DIR = "jobtypes";
+  private static final String AZ_DIR = "azkaban_libs";
+
+  private static final Logger logger = Logger.getLogger(FlowContainer.class);
+
+  // FlowRunnerManager specific code
+  private final ExecutorService executorService;
+  private final ExecutorLoader executorLoader;
+  private final ProjectLoader projectLoader;
+  private final JobTypeManager jobTypeManager;
+  //private final AzkabanEventReporter azkabanEventReporter;
+  private final Props azKabanProps;
+  private Props globalProps;
+
+  private final File projectDir;
+
+  private KeyStore keyStore;
+  private Future flowFuture;
+
+  // We want to limit the log sizes to about 20 megs
+  private final String jobLogChunkSize;
+  private final int jobLogNumFiles;
+  // If true, jobs will validate proxy user against a list of valid proxy users.
+  private final boolean validateProxyUser;
+
+
+  private FlowContainer(final Path projectDirPath, final Props props)
+          throws IOException {
+    this.projectDir = projectDirPath.toFile();
+
+    // Create Azkaban Props Map
+    this.azKabanProps = props;
+    // Setup global props if applicable
+    final String globalPropsPath = this.azKabanProps.getString("executor.global.properties", null);
+    if (globalPropsPath != null) {
+      this.globalProps = new Props(null, globalPropsPath);
+    }
+
+    // Setup DAO, a lot of it is redundant
+    final DataSource dataSource= new MySQLDataSource(this.azKabanProps,
+        new DBMetrics(new MetricsManager(new MetricRegistry())));
+
+    final DatabaseOperator dbOperator = new DatabaseOperator(
+        new QueryRunner(dataSource));
+
+    this.executorLoader = setupDao(dbOperator);
+
+    // project Loader
+    this.projectLoader = new JdbcProjectImpl(this.azKabanProps, dbOperator);
+
+    // setup executor service, TODO : revisit
+    this.executorService = Executors.newSingleThreadExecutor();
+
+    this.jobLogChunkSize = this.azKabanProps.getString("job.log.chunk.size", "5MB");
+    this.jobLogNumFiles = this.azKabanProps.getInt("job.log.backup.index", 4);
+    this.validateProxyUser = this.azKabanProps.getBoolean("proxy.user.lock.down", false);
+    this.jobTypeManager =
+        new JobTypeManager(
+            this.azKabanProps.getString(AzkabanExecutorServer.JOBTYPE_PLUGIN_DIR,
+                PluginManager.JOBTYPE_DEFAULTDIR),
+            this.globalProps, getClass().getClassLoader());
+    // Fetch keyStore props and use it get the KeyStore, put it in JobTypeManager
+    Props keyStoreLoadProps = this.jobTypeManager.getKeyStoreLoadProps();
+    if (keyStoreLoadProps != null) {
```

Comment:
If keyStoreLoadProps is null, do we want to throw an exception or consume it? If we intend to throw an exception probaly early check like below will be helpful
if (null == keyStoreLoadProps) {
 // throw exception
}